sign compare warning in Py

```
/root/repo/packages/openpmd-api/build/openpmd-api-0.14.4/src/binding/python/RecordComponent.cpp:957:77: warning: comparison of integers of different signs: 'pybind11::ssize_t' (aka 'long') and 'unsigned int' [-Wsign-compare]
            if( offset_in.size() == 1u && offset_in.at(0) == 0u && a.ndim() > 1u )
                                                                   ~~~~~~~~ ^ ~~
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -896,11 +896,11 @@ if(CMAKE_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR)
         #                     /usr/lib/llvm-6.0/lib/clang/6.0.0/lib/linux/libclang_rt.ubsan_minimal-x86_64.so
         # at runtime when used with symbol-hidden code (e.g. pybind11 module)
 
-        set(CMAKE_CXX_FLAGS "-Wall -Wextra -Wpedantic -Wshadow -Woverloaded-virtual -Wextra-semi -Wunreachable-code ${CMAKE_CXX_FLAGS}")
+        set(CMAKE_CXX_FLAGS "-Wall -Wextra -Wpedantic -Wshadow -Woverloaded-virtual -Wextra-semi -Wunreachable-code -Wsign-compare ${CMAKE_CXX_FLAGS}")
     elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
         set(CMAKE_CXX_FLAGS "-w3 -wd193,383,1572 ${CMAKE_CXX_FLAGS}")
     elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
-        set(CMAKE_CXX_FLAGS "-Wall -Wextra -Wpedantic -Wshadow -Woverloaded-virtual -Wunreachable-code ${CMAKE_CXX_FLAGS}")
+        set(CMAKE_CXX_FLAGS "-Wall -Wextra -Wpedantic -Wshadow -Woverloaded-virtual -Wunreachable-code -Wsign-compare ${CMAKE_CXX_FLAGS}")
     elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
         # Warning C4503: "decorated name length exceeded, name was truncated"
         # Symbols longer than 4096 chars are truncated (and hashed instead)

--- a/src/binding/python/RecordComponent.cpp
+++ b/src/binding/python/RecordComponent.cpp
@@ -953,7 +953,7 @@ void init_RecordComponent(py::module &m) {
             // default arguments
             //   offset = {0u}: expand to right dim {0u, 0u, ...}
             Offset offset = offset_in;
-            if( offset_in.size() == 1u && offset_in.at(0) == 0u && a.ndim() > 1u )
+            if( offset_in.size() == 1u && offset_in.at(0) == 0u && a.ndim() > 1 )
                 offset = Offset(a.ndim(), 0u);
 
             //   extent = {-1u}: take full size

